CRM/Dedupe/MergeHandler.php

```diff
@@ -213,6 +213,79 @@ public function copyDataToNewBlockDAO($otherBlockId, $name, $blkCount) {
     return $otherBlockDAO;
   }
 
+  /**
+   * Get blocks, if any, to update for the deleted contact.
+   *
+   * If the deleted contact no longer has a primary address but still has
+   * one or more blocks we want to ensure the remaining block is updated
+   * to have is_primary = 1 in case the contact is ever undeleted.
+   *
+   * @param string $entity
+   *
+   * @return array
+   * @throws \CRM_Core_Exception
+   */
+  public function getBlocksToUpdateForDeletedContact($entity) {
+    $movedBlocks = $this->getLocationBlocksToMerge()[$entity];
+    $deletedContactsBlocks = $this->getLocationBlocksForContactToRemove()[$entity];
+    $unMovedBlocks = array_diff_key($deletedContactsBlocks, $movedBlocks);
```

Comment:
If this line is updated to 

    $unMovedBlocks = array_values(array_diff_key($deletedContactsBlocks, $movedBlocks));

it passes the above testing.

Reply:
Thanks @jitendrapurohit - I've just made that change - if it still passes tests then the flow I tested will still  be fine